tests/components/reolink/test_media_source.py

```diff
@@ -0,0 +1,283 @@
+
+    browse = await async_browse_media(
+        hass, f"{URI_SCHEME}{DOMAIN}/{browse_res_main_id}"
+    )
+
+    browse_days_id = f"DAYS|{config_entry.entry_id}|{TEST_CHANNEL}|{TEST_STREAM}"
+    browse_day_0_id = f"DAY|{config_entry.entry_id}|{TEST_CHANNEL}|{TEST_STREAM}|{TEST_YEAR}|{TEST_MONTH}|{TEST_DAY}"
+    browse_day_1_id = f"DAY|{config_entry.entry_id}|{TEST_CHANNEL}|{TEST_STREAM}|{TEST_YEAR}|{TEST_MONTH}|{TEST_DAY2}"
+    assert browse.domain == DOMAIN
+    assert browse.title == f"{TEST_NVR_NAME} High res."
+    assert browse.identifier == browse_days_id
+    assert browse.children[0].identifier == browse_day_0_id
+    assert browse.children[1].identifier == browse_day_1_id
+
+    # browse camera recording files on day
+    mock_vod_file = MagicMock()
+    mock_vod_file.start_time = datetime(
+        TEST_YEAR, TEST_MONTH, TEST_DAY, TEST_HOUR, TEST_MINUTE
+    )
+    mock_vod_file.duration = timedelta(minutes=15)
+    mock_vod_file.file_name = TEST_FILE_NAME
+    reolink_connect.request_vod_files.return_value = ([mock_status], [mock_vod_file])
+
+    browse = await async_browse_media(hass, f"{URI_SCHEME}{DOMAIN}/{browse_day_0_id}")
+
+    browse_files_id = f"FILES|{config_entry.entry_id}|{TEST_CHANNEL}|{TEST_STREAM}"
+    browse_file_id = (
+        f"FILE|{config_entry.entry_id}|{TEST_CHANNEL}|{TEST_STREAM}|{TEST_FILE_NAME}"
+    )
+    assert browse.domain == DOMAIN
+    assert (
+        browse.title == f"{TEST_NVR_NAME} High res. {TEST_YEAR}/{TEST_MONTH}/{TEST_DAY}"
+    )
+    assert browse.identifier == browse_files_id
+    assert browse.children[0].identifier == browse_file_id
+
+
+async def test_browsing_unsupported_encoding(
+    hass: HomeAssistant,
+    reolink_connect: MagicMock,
+    config_entry: MockConfigEntry,
+) -> None:
+    """Test browsing a Reolink camera with unsupported stream encoding."""
+    with patch("homeassistant.components.reolink.PLATFORMS", [Platform.CAMERA]):
+        assert await hass.config_entries.async_setup(config_entry.entry_id) is True
+    await hass.async_block_till_done()
+
+    browse_root_id = f"CAM|{config_entry.entry_id}|{TEST_CHANNEL}"
+
+    # browse resolution select/camera recording days when main encoding unsupported
+    mock_status = MagicMock()
+    mock_status.year = TEST_YEAR
+    mock_status.month = TEST_MONTH
+    mock_status.days = (TEST_DAY, TEST_DAY2)
+    reolink_connect.request_vod_files.return_value = ([mock_status], [])
+    reolink_connect.time.return_value = None
+    reolink_connect.get_encoding.return_value = "h265"
+
+    browse = await async_browse_media(hass, f"{URI_SCHEME}{DOMAIN}/{browse_root_id}")
+
+    browse_days_id = f"DAYS|{config_entry.entry_id}|{TEST_CHANNEL}|sub"
+    browse_day_0_id = f"DAY|{config_entry.entry_id}|{TEST_CHANNEL}|sub|{TEST_YEAR}|{TEST_MONTH}|{TEST_DAY}"
+    browse_day_1_id = f"DAY|{config_entry.entry_id}|{TEST_CHANNEL}|sub|{TEST_YEAR}|{TEST_MONTH}|{TEST_DAY2}"
```

Comment:
Might be a bit more readable if you used a local var for `entry_id`

Reply:
done